### PR TITLE
feat: add referral set referee X day aggregate totals to API results

### LINKED ISSUE #9541: [Feature]: Finalise referral set API (pt2)
# Feature Overview

This is the second part of the front end request for referral set APIs: https://github.com/vegaprotocol/vega/issues/9261

**In order to** support front end 
**We will** ensure the new referral set API covers all the below
**So that** front end can build
- The referrer stats on page 10 of the wireframes (and in the miro that will soon exist [here](https://miro.com/app/board/uXjVOs_PBKk=/?share_link_id=783456245937))

## Specs
https://github.com/vegaprotocol/specs/blob/cosmicelevator/protocol/0083-RFPR-on_chain_referral_program.md#apis

## Tasks
A checklist of the tasks that are needed to develop the feature and meet the acceptance criteria and feature test scenarios.
- [x] Ensure the referral set API covers all of the remaining details in the spec
  - [x] Members list
    - [x] Total volume in last 30 days (in quantum presumably since that is what is going into the overall set calc)
    - [x] Commission earned from party in last 30 days (if possible)

### PR DESCRIPTION
**Pull request by guoguojin**
Closes #9541

### CODE REVIEW

File: datanode/gateway/graphql/schema.graphql [hunk 1]
```diff
@@ -6056,6 +6058,10 @@ type ReferralSetReferee {
   joinedAt: Timestamp!
   "Epoch in which the party joined the set."
   atEpoch: Int!
+  "Total volume of the referee's taker trades over the aggregation period."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6061]:
```suggestion
  "Total notional volume of the referee's aggressive trades over the aggregation period, default is 30 days."
```

File: datanode/gateway/graphql/schema.graphql [hunk 2]
```diff
@@ -6056,6 +6058,10 @@ type ReferralSetReferee {
...
+  TotalRefereeNotionalTakerVolume: String!
+  "Total rewards generated from the referee over the aggregation period."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:6063]:
```suggestion
  "Total rewards generated from the referee over the aggregation period, default is 30 days."
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 3]
```diff
@@ -3734,6 +3734,10 @@ message ReferralSetReferee {
   int64 joined_at = 3;
   // Epoch at which the party joined the set.
   uint64 at_epoch = 4;
+  // Total volume generated by the referee over the aggregation period.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:3737]:
```suggestion
  // Total notional volume of the referee's aggressive trades over the aggregation period, default is 30 days.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -3734,6 +3734,10 @@ message ReferralSetReferee {
...
+  string total_referee_notional_taker_volume = 5;
+  // Total rewards generated by the referee over the aggregation period.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:3739]:
```suggestion
  // Total rewards generated by the referee over the aggregation period, default is 30 days.
```

**Review by EVODelavega — DISMISSED** [review]

**Review by candida-d — DISMISSED** [review]

File: datanode/sqlstore/referral_sets.go [hunk 5]
```diff
@@ -248,3 +247,28 @@ func (rs *ReferralSets) ListReferralSetReferees(ctx context.Context, referralSet
...
+    and jsonb_typeof(referrer_rewards_generated) != 'null'
+    group by gen_rewards->>'party'
+)
+SELECT rf.referral_set_id, rf.referee, rf.joined_at, rf.at_epoch, rf.vega_time, coalesce(pv.period_volume, 0) period_volume, coalesce(pr.period_rewards, 0) period_rewards_paid
+from referral_set_referees rf
+join referral_sets rs on rf.referral_set_id = rs.id
+left join ref_period_volume pv on rf.referee = pv.party
+left join ref_period_rewards pr on rf.referee = pr.party
+	`, aggregationDays, aggregationDays)
+}
```
Comment (ValentinTrinque) [inline datanode/sqlstore/referral_sets.go:274]: https://www.youtube.com/watch?v=8sCApmpxFw4

File: datanode/gateway/graphql/schema.graphql [hunk 6]
```diff
@@ -6056,6 +6058,10 @@ type ReferralSetReferee {
   joinedAt: Timestamp!
   "Epoch in which the party joined the set."
   atEpoch: Int!
+  "Total notional volume of the referee's aggressive trades over the aggregation period, default is 30 days."
+  TotalRefereeNotionalTakerVolume: String!
+  "Total rewards generated from the referee over the aggregation period, default is 30 days."
+  TotalRefereeGeneratedRewards: String!
```
Comment (edd) [inline datanode/gateway/graphql/schema.graphql:6064]: **nitpick (blocking)**:  lower case first letter for consistency

File: datanode/gateway/graphql/schema.graphql [hunk 7]
```diff
@@ -6056,6 +6058,10 @@ type ReferralSetReferee {
   joinedAt: Timestamp!
   "Epoch in which the party joined the set."
   atEpoch: Int!
+  "Total notional volume of the referee's aggressive trades over the aggregation period, default is 30 days."
+  TotalRefereeNotionalTakerVolume: String!
```
Comment (edd) [inline datanode/gateway/graphql/schema.graphql:6062]: **nitpick (blocking)**:  lower case first letter for consistency

File: datanode/gateway/graphql/schema.graphql (hunk 6 above)
Comment (edd) [inline datanode/gateway/graphql/schema.graphql:6064]:
```suggestion
  totalRefereeGeneratedRewards: String!
```

**Review by candida-d — APPROVED** [review]

**Review by EVODelavega — APPROVED** [review]

**Review by asiaznik — APPROVED** [review]

**Pull request merged**